### PR TITLE
Bank account observer

### PR DESCRIPTION
**Pull request by oshkbello**
Please let's review this together. I figured somethings out but wasn't sure what you were trying to achieve with the subaccount methods at some point. So I made assumptions in order to move forward with the task.

### CODE REVIEW

File: app/Observers/BankAccountObserver.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+class BankAccountObserver
+{
+  /**
+   * Handle the bank account "created" event.
+   *
+   * @param  \App\BankAccount  $bankAccount
+   * @return void
+   */
+  public function created(BankAccount $bankAccount)
```
Comment (purplenimbus) [inline app/Observers/BankAccountObserver.php:15]:
```suggestion
  public function created(BankAccount $bank_account)
```
lets use underscores for variable names

File: app/Observers/BankAccountObserver.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    }
+  }
+
+  private function updateSubAccount(BankAccount $bankAccount){
+    if ($bankAccount->tenant->payment_details) {
+      $bankAccount->tenant->updateOrCreateSubAccount();
+    }
+  }
+
+  private function newSubAccount(BankAccount $bankAccount){
```
Comment (purplenimbus) [inline app/Observers/BankAccountObserver.php:92]:
```suggestion
  private function createSubAccount(BankAccount $bank_account){
```

**Comment by oshkbello** [comment]
Since we have moved on from this PR, whats the right thing to do with it?

**Comment by purplenimbus** [comment]
> Since we have moved on from this PR, whats the right thing to do with it?

lets just close it

**Pull request closed without merging**